### PR TITLE
Week2 development

### PR DESCRIPTION
**Pull request by rahulvit2010**
i have added api documentation for product service and i also added liquibase configuration

### CODE REVIEW

File: productservice/src/main/java/com/epam/productservice/controller/ProductRestController.java [hunk 1]
```diff
@@ -42,6 +47,11 @@
 
 
     @PostMapping("/products")
+    @ApiOperation(value = "it Create new Product entry ", response = ResponseEntity.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "CREATED"),
+            @ApiResponse(code = 406, message = "NOT_ACCEPTABLE"),
```
Comment (vinaylakkam) [inline productservice/src/main/java/com/epam/productservice/controller/ProductRestController.java]: Are you throwing 406? If not please remove it.

File: productservice/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                  type: varchar(255)
+                  constraints:
+                     nullable: false
+              - column:
+                  name: product_id
+                  type: int
+                  constraints:
+                     nullable: false
+                     foreignKeyName: fk_product_id
+                     references: product(id)
```
Comment (vinaylakkam) [inline productservice/src/main/resources/db/changelog/db.changelog-master.yaml]: This `prodcut_review` should be in different repo no?

File: productservice/src/main/java/com/epam/productservice/controller/ProductRestController.java (hunk 1 above)
Comment (rahulvit2010) [reply productservice/src/main/java/com/epam/productservice/controller/ProductRestController.java]: i was throwing it when user where giving invalid category, but i think in this case i should return error code 400(BAD_REQUEST. i will remove and update incontroller

File: productservice/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 2 above)
Comment (rahulvit2010) [reply productservice/src/main/resources/db/changelog/db.changelog-master.yaml]: yes vinay...i made this yaml with my existing datamodel.....i was having review linked to product......

Comment (vinaylakkam) [reply productservice/src/main/resources/db/changelog/db.changelog-master.yaml]: review should be a different database instance. Each micro service needs to have its own database. Hope you got this

Comment (rahulvit2010) [reply productservice/src/main/resources/db/changelog/db.changelog-master.yaml]: i got your point vinay

**Pull request merged**